pkg/game/scoring_roads_test.go

```diff
@@ -0,0 +1,228 @@
+				t.Fatalf("placing tile number: %#v failed. expected %+v meeples for player %v, got %+v instead", i, expectedMeeples[i], playerID, report.ReturnedMeeples[playerID])
+			}
+		}
+	}
+}
+
+/*
+Test loop, but the final tile is a crossRoad
+Roads:
+  - 0 -
+    1 -	2
+    |	|
+  - 4 -	3
+*/
+func TestBoardScoreRoadLoopCrossroad(t *testing.T) {
+	var report elements.ScoreReport
+	var boardInterface interface{} = NewBoard(tilesets.StandardTileSet())
+	board := boardInterface.(*board)
+
+	tiles := []elements.PlacedTile{
+		test.GetTestCustomPlacedTile(tiletemplates.RoadsTurn().Rotate(3), 1),
+		test.GetTestCustomPlacedTile(tiletemplates.RoadsTurn().Rotate(0), 2),
+		test.GetTestCustomPlacedTile(tiletemplates.RoadsTurn().Rotate(1), 1),
+
+		test.GetTestCustomPlacedTile(tiletemplates.TCrossRoad().Rotate(2), 1),
+	}
+
+	// add meeple to first road
+	tiles[0].GetPlacedFeatureAtSide(side.Right).Meeple.PlayerID = 1
+	tiles[0].GetPlacedFeatureAtSide(side.Right).Meeple.MeepleType = elements.NormalMeeple
+
+	// set positions
+	tiles[0].Position = elements.NewPosition(0, -1)
+	tiles[1].Position = elements.NewPosition(1, -1)
+	tiles[2].Position = elements.NewPosition(1, -2)
+	tiles[3].Position = elements.NewPosition(0, -2)
+
+	expectedScores := []uint32{0, 0, 0, 4}
+	expectedMeeples := [][]uint8{nil, nil, nil, {0, 1}}
+
+	// --------------- Placing tile ----------------------
+
+	for i := range 4 {
+		println(i, ":")
+		_, err := board.PlaceTile(tiles[i])
+		if err != nil {
+			t.Fatalf("error placing tile number: %#v ", i)
+		}
+		report = board.ScoreRoads(tiles[i])
+
+		if report.ReceivedPoints[1] != expectedScores[i] {
+			t.Fatalf("placing tile number: %#v failed. expected %+v for player %v, got %+v instead", i, expectedScores[i], 1, report.ReceivedPoints[1])
+		}
+
+		if !reflect.DeepEqual(report.ReturnedMeeples[1], expectedMeeples[i]) {
+			t.Fatalf("placing tile number: %#v failed. expected %+v meeples for player %v, got %+v instead", i, expectedMeeples[i], 1, report.ReturnedMeeples[1])
+		}
+
+	}
+}
+
+/*
+Test crossroad end to monastery and starting tile between
+
+Roads:
+
+	1 -	0 -	2
+*/
+func TestBoardScoreRoadCityMonastery(t *testing.T) {
+	var report elements.ScoreReport
+	var boardInterface interface{} = NewBoard(tilesets.StandardTileSet())
+	board := boardInterface.(*board)
+
+	tiles := []elements.PlacedTile{
+		test.GetTestCustomPlacedTile(tiletemplates.MonasteryWithSingleRoad().Rotate(3), 1), // monastery to right
+		test.GetTestCustomPlacedTile(tiletemplates.MonasteryWithSingleRoad().Rotate(1), 1), // monastery to left
+	}
+
+	// add meeple to first road
+	tiles[0].GetPlacedFeatureAtSide(side.Right).Meeple = elements.Meeple{PlayerID: 1, MeepleType: elements.NormalMeeple}
+
+	// set positions
+	tiles[0].Position = elements.NewPosition(-1, 0)
+	tiles[1].Position = elements.NewPosition(1, 0)
+
+	expectedScores := []uint32{0, 3}
+	expectedMeeples := [][]uint8{nil, {0, 1}} // expected zero Nones, and 1 normal meeple
```

Comment:
Why do we need to track the amount of "None" meeples in score report?

Reply:
Actually i didn't know how to create map without the None meeple key. I create the map in score_report.go in line 90.  Is there better way to achieve this? 
There is something like this:
scoreReport.ReturnedMeeples[uint8(meeple.PlayerID)] = []uint8{0, 0}

When i did it with one zero, it didn't work.

Reply:
ReturnedMeeples field is defined as map of slices, thus slice with single value should be valid.

Reply:
Yea, but NormalMeeple has value 1, and NoneMeeple 0. Thus checking how many certain meeple type did player have, requires substracting by one, 
F.e: 
scoreReport.ReturnedMeeples[uint8(meeple.PlayerID)][meeple.MeepleType-1]++
instead of
scoreReport.ReturnedMeeples[uint8(meeple.PlayerID)][meeple.MeepleType]++

which would be obscure.  I'd rather change it to map of maps, to avoid it, but still in different pull request.

Reply:
I'm facing the same problem now and I'd also prefer the non-obscure approach of just having a 0 at the first index to allow easy ignoring of None meeples